src/config.rs

```diff
@@ -0,0 +1,57 @@
+use crate::limit::Limit;
+
+use std::time::Duration;
+
+/// The configuration that specifies how to run the benchmarks.
+#[derive(Debug)]
+pub(crate) struct Config {
+    /// The path of the result file.
+    pub result_path: String,
+    /// Run the benchmarks in quick mode (for development/testing purposes).
+    pub quick: bool,
+    /// Don't actually run the benchmarks (for development/testing purposes).
+    pub dry_run: bool,
+    /// Automatically reboot between pexecs.
+    pub reboot: bool,
+    /// The list of emails to send notifications/errors to.
+    pub mail_to: Vec<String>,
+    /// The number of in-process iterations.
+    pub in_proc_iters: usize,
+    /// The number of process executions.
+    pub pexecs: usize,
+    /// The stack size limit. `None` by default.
+    pub stack_lim: Option<Limit>,
+    /// The heap size limit. `None` by default.
+    pub heap_lim: Option<Limit>,
+    /// The amount of time to wait before taking the initial temperature reading.
+    pub temp_read_pause: Duration,
+    /// CPU pinning. `false` by default.
+    pub enable_pinning: bool,
+    /// Include wallclock times in the results. `true` by default.
+    pub wallclock: bool,
+    /// Include the number of CPU cycle counts in the result. `true` by default.
+    pub core_cycles: bool,
+    /// Include aperf/mperf counts in the result. `true` by default.
+    pub perf: bool,
```

Comment:
I wonder if we should start with a minimal set of config options and add more as we implement them. We've not yet decided how many of the Krun features we actually want to implement.

Reply:
Sure, that's a fair point.